### DISCUSSION TITLE
Remove tool for inpaint (similar to photoshop remove tool)

### DISCUSSION THREAD

**Discussion by pad0n** [lllyasviel/Fooocus#1763] [Ideas]
Hello everyone,

I believe the title of the topic is self-explanatory. It would be incredibly beneficial to have the ability to delete objects from photos by simply selecting them and pressing "delete." A similar function is available in Photoshop and also in Stylar AI.

I've attempted something similar using Fooocus, but it's not as straightforward as I hoped. The challenge arises when trying to precisely select an object. For instance, when attempting to remove an object (let's say a tail on a cow - sometimes two tails get generated, and selecting just one tail can be tricky) writing a prompt for what should replace the object is a real challenge. If the tail is against the backdrop of the sky, it's relatively easy to prompt "sky" for removal. However, when dealing with a complex scene involving multiple cows, pigs, wheat, other people, and so forth, it becomes incredibly challenging to generate the right prompt for removing just that specific object without generating unnecessary elements. Also i tried leaving the input field blank but then usually the tail didn't change at all or another similar tail was generated.

And yes, of course I can generate the image, save it, go to photoshop, delete the desired element, but these are additional redundant actions. I'm just suggesting to add one more interesting feature, as other AIs are already starting to offer them in their arsenal (stylar for example)

**Comment by docppp** [comment]
It can be achieved with Inpaint tool. For example if you have a dog on a bench, you can select the dog and prompt "empty bench". It was even showed in some topic in the Discussion section, but I can find it now.

**Reply by pad0n** [reply]
Sure, I understand that, and that's why I further described situations where such a description didn’t work well — when there was a very complex scene on the background with a complete frenzy happening in the background. In the original message, it starts from 'However, when dealing with a complex scene involving...' :)
